applications/sdp/mspi/src/main.c

```diff
@@ -0,0 +1,220 @@
+	}
+
+	/* Send data */
+	xfer_ll_params.ce_hold = false;
+	xfer_ll_params.word_size = 32;
+	/* TODO: this system needs to be fixed as for now,
+	 * there are problems when (data_length%4) != 0
+	 */
+	xfer_ll_params.data_len = WORDS_FOR_BYTES(data_length);
+	xfer_ll_params.data_to_send = data_buffer + XFER_DATA_IDX;
+
+	if (mspi_dev_configs.io_mode == MSPI_IO_MODE_SINGLE) {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
+	} else {
+		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
+	}
+}
+
+__attribute__((interrupt)) void hrt_handler_write_single(void)
+{
+	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+
+	write_single_by_word(xfer_ll_params);
+}
+
+__attribute__((interrupt)) void hrt_handler_write_quad(void)
+{
+	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+
+	write_quad_by_word(xfer_ll_params);
+}
+
+int main(void)
+{
+	IRQ_DIRECT_CONNECT(HRT_VEVIF_IDX_WRITE_SINGLE, HRT_IRQ_PRIORITY, hrt_handler_write_single,
+			   0);
+	nrf_vpr_clic_int_enable_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE), true);
+
+	IRQ_DIRECT_CONNECT(HRT_VEVIF_IDX_WRITE_QUAD, HRT_IRQ_PRIORITY, hrt_handler_write_quad, 0);
+	nrf_vpr_clic_int_enable_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD), true);
+
+	nrf_vpr_csr_rtperiph_enable_set(true);
+
+	/* This pin initialization is temporary until code is merged with app with GPIO
+	 * initialization
+	 */
+	nrf_gpio_pin_dir_t dir = NRF_GPIO_PIN_DIR_OUTPUT;
+	nrf_gpio_pin_input_t input = NRF_GPIO_PIN_INPUT_DISCONNECT;
+	nrf_gpio_pin_pull_t pull = NRF_GPIO_PIN_NOPULL;
+	nrf_gpio_pin_drive_t drive = NRF_GPIO_PIN_E0E1;
+
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, SCLK_PIN), &dir, &input, &pull, &drive, NULL);
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, D0_PIN), &dir, &input, &pull, &drive, NULL);
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, D1_PIN), &dir, &input, &pull, &drive, NULL);
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, D2_PIN), &dir, &input, &pull, &drive, NULL);
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, D3_PIN), &dir, &input, &pull, &drive, NULL);
+	nrfy_gpio_reconfigure(NRF_GPIO_PIN_MAP(2, CS_PIN), &dir, &input, &pull, &drive, NULL);
+
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, SCLK_PIN), NRF_GPIO_PIN_SEL_VPR);
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, D0_PIN), NRF_GPIO_PIN_SEL_VPR);
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, D1_PIN), NRF_GPIO_PIN_SEL_VPR);
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, D2_PIN), NRF_GPIO_PIN_SEL_VPR);
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, D3_PIN), NRF_GPIO_PIN_SEL_VPR);
+	nrfy_gpio_pin_control_select(NRF_GPIO_PIN_MAP(2, CS_PIN), NRF_GPIO_PIN_SEL_VPR);
+
+	configure_clock(MSPI_CPP_MODE_0);
+
+	uint16_t direction = nrf_vpr_csr_vio_dir_get();
+
+	nrf_vpr_csr_vio_dir_set(direction | PIN_DIR_OUT_MASK(CS_PIN));
+
+	uint16_t output = nrf_vpr_csr_vio_out_get();
```

Comment:
variable declarations to top of function

Reply:
fixed